power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -27,6 +28,26 @@ namespace power_grid_model::meta_data {
 constexpr auto const& as_array(msgpack::object const& obj) { return obj.via.array; }
 constexpr auto const& as_map(msgpack::object const& obj) { return obj.via.map; }
 // NOLINTEND(cppcoreguidelines-pro-type-union-access)
+
+constexpr void unpack(msgpack::object const& obj, std::floating_point auto& attr) {
+    using namespace msgpack;
+    using namespace std::string_view_literals;
+
+    auto infinity = std::numeric_limits<double>::infinity();
+    if (obj.type == type::STR) {
+        auto const& obj_string = obj.as<std::string_view>();
+        if (obj_string == "inf" || obj_string == "+inf") {
+            attr = infinity;
+        } else if (obj_string == "-inf") {
+            attr = -infinity;
+        } else {
+            throw type_error();
+        }
+    } else {
+        obj >> attr;
+    }
+}
```

Comment:
as i said in one of my previous comments, please switch the logic around: default should be parsing the floating point, fallback should be parsing infinity

please also consider using a return statement rather than passing by reference. especially for floating point types, it's actually more efficient to pass by value than to pass by reference